documentation/modules/managing/con-producer-config-properties.adoc

```diff
@@ -44,35 +44,68 @@ Compression is useful for improving throughput and reducing the load on storage,
 
 The `acks=all` setting offers the strongest guarantee of delivery, but it will increase the latency between the producer sending a message and receiving acknowledgment.
-If you don't require such strong guarantees, a setting of `acks=0` or `acks=1` provides either no delivery guarantees or only acknowledgment that the leader partition has written the record to its log.
+If you don't require such strong guarantees, a setting of `acks=0` or `acks=1` provides either no delivery guarantees or only acknowledgment that the leader replica has written the record to its log.
 
-The number of brokers which need to have appended the messages to their logs before the acknowledgment is sent to the producer is determined by the topic's `min.insync.replicas` configuration.
-A typical starting point is to have a topic replication factor of 3, with two in-sync replicas on other brokers.
-In this configuration, the producer can continue unaffected if a single broker is unavailable.
-If a second broker becomes unavailable, the producer won’t receive acknowledgments and won’t be able to produce more messages.
+With `acks=all`, the leader waits for all in-sync replicas to acknowledge message delivery.
+A topic's `min.insync.replicas` configuration sets the minimum required number of in-sync replica acknowledgements.
+The number of acknowledgements include that of the leader and followers. 
 
-.Topic configuration to support `acks=all`
-[source,env]
-----
-# ...
-min.insync.replicas=2 <1>
-# ...
-----
-<1> Use `2` in-sync replicas. The default is `1`.
+A typical starting point is to use the following configuration:
+
+* Producer configuration:
+** `acks=all` (default)
+* Broker configuration for topic replication:
+** `default.replication.factor=3` (default = `1`)
+** `min.insync.replicas=2` (default = `1`)
+
+When you create a topic, you can override the default replication factor.
+You can also override `min.insync.replicas` at the topic level in the topic configuration. 
+
+Strimzi uses this configuration in the example configuration files for multi-node deployment of Kafka. 
+
+The following table describes how this configuration operates depending on the availability of followers that replicate the leader replica.
+
+.Follower availability
+[cols="2,3,2",options="header"]
+|===
+
+|Number of followers available and in-sync
+|Acknowledgements 
+|Producer can send messages?
+
+|2
+|The leader waits for 2 follower acknowledgements
+|Yes
+
+|1
+|The leader waits for 1 follower acknowledgement
+|Yes
+
+|0
+|The leader raises an exception
+|No
+
+|===
+
+A topic replication factor of 3 creates one leader replica and two followers.
+In this configuration, the producer can continue if a single follower is unavailable.
+Some delay can occur whilst a failed broker is removed from the in-sync replicas or a new leader is created.
+If a second follower becomes unavailable, the producer won’t receive acknowledgments and won’t be able to produce more messages.
```

Comment:
I don't think it is a wrong, but I had the impression as if there was no response from the leader. If a second replica broker becomes unavailable, the producer won’t receive acknowledgments **`to be considered successful`** from the leader. 

FYI, the leader will respond error code for NotEnoughReplicas or NotEnoughReplicasAfterAppend. And the producer with 'acks=all' will raise either NotEnoughReplicas or NotEnoughReplicasAfterAppend exception on the producer side. And those exception are retryable exceptions, so the producer can repeat retries if there is retry configuration.

> [Kafka doc min.insync.replicas] https://kafka.apache.org/documentation/
> When a producer sets acks to "all" (or "-1"), min.insync.replicas specifies the minimum number of replicas that must acknowledge a write for the write to be considered successful. If this minimum cannot be met, then the producer will raise an exception (either NotEnoughReplicas or NotEnoughReplicasAfterAppend).

Reply:
It might be worth saying something on the error/exception in a general way. Updated as follows:

> A topic replication factor of 3 creates one leader replica and two followers. In this configuration, the producer can continue if a single follower is unavailable. Some delay can occur whilst removing a failed broker from the in-sync replicas or a creating a new leader. If the second follower is also unavailable, message delivery will not be successful. Instead of acknowledging successful message delivery, the leader sends an error (_not enough replicas_) to the producer. The producer raises an equivalent exception. With retries configuration, the producer can resend the failed message request.

Reply:
LGTM. Thank you for updating.